web-common/src/features/scheduled-reports/ScheduledReportDialog.svelte

```diff
@@ -52,124 +47,102 @@
-          project,
-          name: reportName,
-          data: {
-            options: {
-              displayName: values.title,
-              refreshCron: refreshCron, // for testing: "* * * * *"
-              refreshTimeZone: values.timeZone,
-              queryName: reportSpec?.queryName ?? "MetricsViewAggregation",
-              queryArgsJson: JSON.stringify(
-                reportSpec?.queryArgsJson
-                  ? JSON.parse(reportSpec.queryArgsJson)
-                  : queryArgs,
-              ),
-              exportLimit: values.exportLimit || undefined,
-              exportFormat: values.exportFormat,
-              emailRecipients: values.recipients
-                .map((r) => r.email)
-                .filter(Boolean),
-              webOpenState: reportSpec
-                ? (reportSpec.annotations as V1ReportSpecAnnotations)[
-                    "web_open_state"
-                  ]
-                : metricsViewProto,
-              webOpenPath: exploreName ? `/explore/${exploreName}` : undefined,
-            },
+  ) as ValidationAdapter<ReportValues>;
+
+  async function handleSubmit(values: ReportValues) {
+    const refreshCron = convertFormValuesToCronExpression(
+      values.frequency,
+      values.dayOfWeek,
+      values.timeOfDay,
+    );
+
+    try {
+      await $mutation.mutateAsync({
+        organization,
+        project,
+        name: reportName,
+        data: {
+          options: {
+            displayName: values.title,
+            refreshCron: refreshCron, // for testing: "* * * * *"
+            refreshTimeZone: values.timeZone,
+            queryName: reportSpec?.queryName ?? "MetricsViewAggregation",
+            queryArgsJson: JSON.stringify(
+              reportSpec?.queryArgsJson
+                ? JSON.parse(reportSpec.queryArgsJson)
+                : queryArgs,
+            ),
+            exportLimit: values.exportLimit || undefined,
+            exportFormat: values.exportFormat,
+            emailRecipients: values.emailRecipients.filter(Boolean),
+            slackChannels: values.enableSlackNotification
+              ? values.slackChannels.filter(Boolean)
+              : undefined,
+            slackUsers: values.enableSlackNotification
+              ? values.slackUsers.filter(Boolean)
+              : undefined,
+            webOpenState: reportSpec
+              ? (reportSpec.annotations as V1ReportSpecAnnotations)[
+                  "web_open_state"
+                ]
+              : metricsViewProto,
+            webOpenPath: exploreName ? `/explore/${exploreName}` : undefined,
           },
-        });
-
-        if (reportSpec) {
-          await queryClient.invalidateQueries(
-            getRuntimeServiceGetResourceQueryKey($runtime.instanceId, {
-              "name.name": reportName,
-              "name.kind": ResourceKind.Report,
-            }),
-          );
-        }
+        },
+      });
 
+      const isEdit = !!reportSpec;
+      if (isEdit) {
```

Comment:
For clarity, this `isEdit` variable could be used in more places throughout this file – wherever there's a ternary based on `reportSpec`